src/main/java/com/linecorp/armeria/common/http/HttpObject.java

```diff
@@ -19,4 +19,12 @@
 /**
  * The common interface for HTTP/2 message objects, {@link HttpHeaders} and {@link HttpData}.
  */
-public interface HttpObject {}
+public interface HttpObject {
+
+    /**
+     * Gets whether the stream should be ended when writing this object. This can be useful for
+     * {@link HttpHeaders}-only responses or to more efficiently close the stream along with the last piece of
+     * {@link HttpData}.
```

Comment:
In that case, we would send an empty frame that closes the stream though right? This comment is about what happens when "writing this object", which I think even in that case is correctly stated. Does that make sense?